packages/core/src/transport.ts

```diff
@@ -132,8 +133,12 @@ export class Batch<T> {
   }
 
   private sizeInBytes(candidate: string) {
-    // tslint:disable-next-line no-bitwise
-    return ~-encodeURI(candidate).split(/%..|./).length
+    // Using this approximate value prevent any costly calculation while validating the byte size requirement
+    if (!this.biggerThanOneByteRegex.test(candidate)) {
+      return candidate.length
+    }
+
+    return candidate.length + 4 * (candidate.match(this.biggerThanOneByteRegex) || '').length
```

Comment:
```js
'é'.match(/([^\u0000-\u007F])/).length // 2
```
so if candidate = 'é', the size in byte will be `1 + 4 * 2 = 9` 

You probably want to have a regex like `/[^\u0000-\u007F]/g`.